pageserver/src/storage_sync/index.rs

```diff
@@ -448,22 +427,26 @@ pub struct IndexLayerMetadata {
 impl From<&'_ LayerFileMetadata> for IndexLayerMetadata {
     fn from(other: &'_ LayerFileMetadata) -> Self {
         IndexLayerMetadata {
-            file_size: other.file_size,
+            file_size: Some(other.file_size),
         }
     }
 }
 
 fn separate_paths_and_metadata(
     timeline_path: &Path,
-    input: &HashMap<PathBuf, LayerFileMetadata>,
+    input: &HashMap<PathBuf, Option<LayerFileMetadata>>,
     output: &mut HashSet<RelativePath>,
     layer_metadata: &mut HashMap<RelativePath, IndexLayerMetadata>,
 ) -> anyhow::Result<()> {
-    for (path, metadata) in input {
+    for (path, local_layer_metadata) in input {
         let rel_path = RelativePath::new(timeline_path, path)?;
-        let metadata = IndexLayerMetadata::from(metadata);
+        // TODO kb wrong transition
```

Comment:
Note: this code has this and lacks the way of "upgrading" the old files with no metadata: we can do this, since download the entire timeline locally.